### PR TITLE
CKV_GCP_103 - Ensure Dataproc Clusters do not have public IPs

### PR DESCRIPTION
**Pull request by jdyke**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

This PR adds a new Checkov check for Public IP Dataproc clusters. By default Dataproc clusters have public IPs.

### CODE REVIEW

File: checkov/terraform/checks/resource/gcp/DataprocPublicIpCluster.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        supported_resources = ['google_dataproc_cluster']
+        categories = [CheckCategories.GENERAL_SECURITY]
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def get_inspected_key(self):
+        return 'cluster_config/[0]/gce_cluster_config/[0]/internal_ip_only'
+
+    # Accounts for if key is present but is set to False
+    def get_expected_value(self):
+        return True
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/gcp/DataprocPublicIpCluster.py:18]: @jdyke This isn't needed as it is the default of the parent

**Review by nimrodkor — APPROVED** [review]
Thanks @jdyke !

File: checkov/terraform/checks/resource/gcp/DataprocPublicIpCluster.py (hunk 1 above)
Comment (jdyke) [reply checkov/terraform/checks/resource/gcp/DataprocPublicIpCluster.py]: Oh weird I thought I had to use that in other examples and the same applied here. I removed it, thanks for catching that

**Review by gruebel — APPROVED** [review]
nice 👍

**Pull request merged**